### ISSUE TITLE
HTTP 409 not properly handled with Exist-DB

### ISSUE THREAD

**Issue by zopyx** [PyFilesystem/webdavfs#9]
```
import fs.opener

dst, _ = fs.opener.open('webdav://admin:xxxx@localhost:6080/exist/webdav/db/sd')

with dst.open(u'index.html', 'wb') as fp:
    fp.write('<hello><xxxx')
```

The webdavfs driver does not handle a HTTP 409 response from ExistDB properly

The code above write an invalid HTML/XML fragment to the database.

Exist-DB check .xml and .html files for XML validity which fails in this case:

```
2017-11-13 15:45:13,257 [eXistThread-1391] ERROR (ExistCollection.java [createFile]:437) - org.xml.sax.SAXParseException: XML document structures must start and end within the same entity. 
```

The result is a HTTP 409 response which is not handled properly:

```

0:0:0:0:0:0:0:1 -  -  [13/Nov/2017:14:45:13 +0000] "PUT /exist/webdav/db/sd/index.html HTTP/1.1" 409 0 
```

Instead the code above passes with out error on the Python level.

**Comment by zopyx** [comment]
This is actually an issue in `webdav2` which does not handle the 409 response properly.

**Comment by zopyx** [comment]
This is fixed with fs==2.0.17
